src/main/java/pl/shop/service/demoShopService.java

```diff
@@ -0,0 +1,89 @@
+package pl.shop.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.shop.exceptions.IncorrectVatException;
+import pl.shop.models.Product;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+
+public class demoShopService {
+
+    /**
+     * To miał być 'właściwy' Service, ale zdążyłem się po drodze nauyczyć, że Spring nie akceptuje
+     * zależności do interfejsów. :/
+     * Ten interfejs nie będzie niezbędny, bo jest tylko na potrzeby Mockowania, ale nie chcę go
+     * wywalać, więc 'Springowy" service przenoszę do klasy Service.
+     */
+
+    // Wstrzykujemy zależność VatProvider. Gdyż jak chcemy coś kupić, musimy płacić VAT <--- top quality
+    // economy knowledge.
+    VatProvider vatProvider;
+
+    public demoShopService(VatProvider vatProvider) {
+        this.vatProvider = vatProvider;
+    }
+
+    /**
+     * Znacie loggera? Każdy go zna, on zawsze fajne pomysły maaaa....
+     * Sorki.
+     * Dorzucony do metod, bo dzięki temu, w bardziej WYSUBLIMOWANY od System.out.println()
+     * sposób, będziemy mieli logowane co się dzieje, co się nie udało i jak bardzo mamy
+     * zacząć być sfrustrowani. Odpalajcie sobie testy i np. tam logger na konsolce będzie
+     * nas informował co się dzieje.
+     */
+    Logger logger = LoggerFactory.getLogger(demoShopService.class);
```

Comment:
https://stackoverflow.com/questions/6653520/why-do-we-declare-loggers-static-final